e2e-tests/global-setup.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { chromium, test as baseTest } from '@playwright/test';
+import { checkEnv } from './helpers/general.js';
+
+import { LoginPage } from './admin/pages/login.js';
+
+async function globalSetup() {
+  await checkEnv([
+    'BOUNDARY_ADDR',
+    'E2E_PASSWORD_ADMIN_LOGIN_NAME',
+    'E2E_PASSWORD_ADMIN_PASSWORD',
+    'E2E_PASSWORD_AUTH_METHOD_ID',
+  ]);
+
+  // Log in and save the authenticated state to reuse in tests
+  const browser = await chromium.launch();
+  const page = await browser.newPage();
+  await page.goto(process.env.BOUNDARY_ADDR);
+
+  const loginPage = new LoginPage(page);
+  await loginPage.login(
+    process.env.E2E_PASSWORD_ADMIN_LOGIN_NAME,
+    process.env.E2E_PASSWORD_ADMIN_PASSWORD,
+  );
+  const storageState = await page
+    .context()
+    .storageState({ path: authenticatedState });
+
+  const state = JSON.parse(storageState.origins[0].localStorage[0].value);
+
+  // Set the token in the environment for use in API requests
+  process.env.E2E_TOKEN = state.authenticated.token;
+
+  await browser.close();
+}
+
+export default globalSetup;
+
+export const authenticatedState = './admin/artifacts/authenticated-state.json';
+
+export const test = baseTest.extend({
```

Comment:
None blocking comment: can we add a brief comment of what is the purpose of extend these vars? i.e: provide local vars to test context.
I think this is nice, I understand it as a "hub" to provide the test context with values that live in environment variables but without having to access env variables constantly.

Reply:
Yup I'll do that

Reply:
Oh thanks for the context! one of the few e2e test PR's I missed!

Thanks for the addition